lib/src/fields/form_builder_signature_pad.dart

```diff
@@ -135,35 +140,50 @@ class _FormBuilderSignaturePadState extends State<FormBuilderSignaturePad> {
                 decoration: BoxDecoration(
                   border: Border.all(color: Colors.grey),
                 ),
-                child: GestureDetector(
-                  onVerticalDragUpdate: (_) {},
-                  child: Signature(
-                    width: widget.width,
-                    height: widget.height,
-                    backgroundColor: widget.backgroundColor,
-                    controller: _effectiveController,
-                  ),
-                ),
+                child: _savedValue != null || _readOnly
+                    ? Card(
+                        color: widget.backgroundColor,
+                        child: Image.memory(
+                          _savedValue,
+                          height: widget.height,
+                          width: widget.width,
+                        ),
+                      )
+                    : GestureDetector(
+                        onVerticalDragUpdate: (_) {},
+                        child: Signature(
+                          width: widget.width,
+                          height: widget.height,
+                          backgroundColor: widget.backgroundColor,
+                          controller: _effectiveController,
+                        ),
+                      ),
               ),
-              Row(
-                children: <Widget>[
-                  Expanded(child: SizedBox()),
-                  FlatButton.icon(
-                    onPressed: () {
-                      _effectiveController.clear();
-                      field.didChange(null);
-                    },
-                    label: Text(
-                      widget.clearButtonText,
-                      style: TextStyle(color: Theme.of(context).errorColor),
-                    ),
-                    icon: Icon(
-                      Icons.clear,
-                      color: Theme.of(context).errorColor,
+              _readOnly
+                  ? const SizedBox(height: 0)
```

Comment:
I recommend `const SizedBox.shrink()` as the best no-op widget.

Reply:
Got your point. Have made the changes as per your suggestion

Reply:
@vin-fandemand Thanks for making this change.  Using _inline conditionals_ can avoid littering the widget hierarchy with _no-op_ widgets, which should lead to better performance.  (I personally also find them easier to read, but the performance issue is why the option is "_better_.")